sklearn/linear_model/_ransac.py

```diff
@@ -357,9 +357,23 @@ def fit(self, X, y, sample_weight=None):
         if self.stop_probability < 0 or self.stop_probability > 1:
             raise ValueError("`stop_probability` must be in range [0, 1].")
 
+        estimator_fit_has_sample_weight = has_fit_parameter(estimator, "sample_weight")
+        estimator_name = type(estimator).__name__
+        if sample_weight is not None and not estimator_fit_has_sample_weight:
+            raise ValueError(
+                "%s does not support sample_weight. Samples"
+                " weights are only used for the calibration"
+                " itself." % estimator_name
+            )
+        sample_weight = _check_sample_weight(sample_weight, X)
+
         if self.residual_threshold is None:
             # MAD (median absolute deviation)
-            residual_threshold = np.median(np.abs(y - np.median(y)))
+            residual_threshold = np.median(
```

Comment:
I don't think that it solves the original issue.

We wanted to impact the loss such that samples do not contribute the same way. Here, we only impact what we should consider as an inlier. Intuitively, if we put weights that discard outliers then the MAD will reduce in your case and we only make it more difficult to get qualified as an inlier. However, we should as well modify the loss.

Reading the original discussion it seems that here we want to have `sample_weight` between 0 and 1 and that we should multiply with `1 - sample_weight` the loss.

Reply:
Hi @glemaitre , I've added a test case that checks residual threshold is calculated correctly with sample weight. 

Meanwhile it seems that modifying loss with `1-sample_weight` will break the test case `test_ransac_fit_sample_weight`, which checks that fitting the model with sample weight `n1, n2, n3` is equal to copy the data `n1, n2, n3` times.